osu.Game/Beatmaps/BeatmapInfo.cs

```diff
@@ -47,6 +47,8 @@ public class BeatmapInfo : RealmObject, IHasGuidPrimaryKey, IBeatmapInfo, IEquat
 
         public BeatmapUserSettings UserSettings { get; set; } = null!;
 
+        public BeatmapUserRank UserRank { get; set; } = null!;
```

Comment:
This just flat out cannot be here. You cannot store *user-dependent* data on `BeatmapInfo`.

Consider that: what if users A and B share a machine, and user A logs out on a local install so that user B can play? They'll *share* their "user rank".